compiler/luci-interpreter/src/core/Tensor.cpp

```diff
@@ -36,6 +36,8 @@ void Tensor::readData(void *data_ptr, size_t data_size) const
 {
   const size_t element_size = getDataTypeSize(element_type());
   const int32_t num_elements = shape().num_elements();
+  if (num_elements <= 0)
```

Comment:
I would instead assert that `data_ptr != nullptr && data_size != 0` after checking `data_size != num_elements * element_size` below. IMHO it is incorrect to call this function for empty tensor.